src/Microsoft.OData.Core/Json/JsonReader.cs

```diff
@@ -285,7 +285,7 @@ public virtual bool Read()
                     }
 
                     // We might see end of array here
-                    if (this.characterBuffer[this.tokenStartIndex] == ']')
+                    if (this.characterBuffer[this.tokenStartIndex] == ']' || this.characterBuffer[this.tokenStartIndex] == ')')
                     {

```

Comment:
Will this also allow me to terminate a JSON array with a closing parens?   i.e., does "names": ["bob","sue") now parse as valid?

Reply:
Great question. The answer is no and I don't have an existing test case for this but I've tested it. I have a test case for the vice-versa scenario (since we currently don't allow bracketed collections in IN expression):

```csharp
public void FilterWithInOperationWithMismatchedClosureCollection()
{
    Action parse = () => ParseFilter("ID in (1,2,3]", HardCodedTestModel.TestModel, HardCodedTestModel.GetPersonType());
    parse.ShouldThrow<ODataException>().WithMessage(ODataErrorStrings.ExpressionLexer_UnbalancedBracketExpression);
}
```
I added the test case that you described with the closing parens and will update the PR after we discuss the other items that you have. No product code changes needed as the parser already takes care of this case.